usaspending_api/references/v2/views/cfda.py

```diff
@@ -35,29 +36,37 @@ def get(self, request, cfda=None):
                 "archived": result["archived"],
                 "forecasted": result["forecasted"],
             }
+
         else:
             response = {"results": CFDA_DICTIONARY.values()}
 
         return Response(response)
 
-    def _populate_cfdas_if_needed(self):
-        global CFDA_DICTIONARY
-        if not CFDA_DICTIONARY:
-            response = self._request_from_grants_api()
-
-            #  grants API is brittle in practice, so if we don't get results retry at a polite rate
-            remaining_tries = 30  # 30 attempts two seconds apart gives the max wait time for the API
-            while not response:
-                if remaining_tries == 0:
-                    raise Exception("Failed to get successful response from Grants API!")
-                sleep(2)
-                response = self._request_from_grants_api()
-                remaining_tries = remaining_tries - 1
-
-            CFDA_DICTIONARY = response
-
-    def _request_from_grants_api(self):
-        return post(
-            "https://www.grants.gov/grantsws/rest/opportunities/search/cfda/totals",
-            headers={"Authorization": f"APIKEY={settings.GRANTS_API_KEY}"},
-        ).json()["cfdas"]
+    def get_cfdas(self):
+        cdfas = {}
+        remaining_tries = 10
+
+        while not cdfas and remaining_tries:
+            remaining_tries -= 1
+            response = post(
+                "https://www.grants.gov/grantsws/rest/opportunities/search/cfda/totals",
+                headers={"Authorization": f"APIKEY={settings.GRANTS_API_KEY}"},
+            )
+            if response.status_code != 200:
+                logger.error(f"Status returned by grants API: {response.status_code}")
```

Comment:
If we're terminating the view business logic here, might as well include the entire response body in the log statement for debugging

Reply:
makes sense; done